service/controller/resource/certificates/resource.go

```diff
@@ -21,11 +20,13 @@ import (
 type Config struct {
 	Name      string
 	Sources   []CertificateSource
-	Target    NameFunc
+	Target    NameFuncWithoutParam
 	K8sClient k8sclient.Interface
 	Logger    micrologger.Logger
 }
 
+type NameFuncWithoutParam func() string
```

Comment:
Add this to be able to remove key.SecretAPICertificates as this basically calls key.Secret without the cluster value